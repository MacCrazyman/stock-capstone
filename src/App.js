import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { NavLink, Outlet } from 'react-router-dom';
import './App.css';
import getCompanies from './app/fetch';
import { loadCompanies } from './redux/reducer';

function App() {
  const dispatch = useDispatch();

  useEffect(async () => {
    const companies = await getCompanies();
    dispatch(loadCompanies(companies));
  }, []);

  return (
    <div className="App">
      <header className="App-header">
        <NavLink to="index"> Home</NavLink>
      </header>
      <main>
        <Outlet />
      </main>
      <footer>
        CC License
      </footer>
    </div>
  );
}

export default App;
