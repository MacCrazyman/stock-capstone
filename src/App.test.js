import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import configureStore from 'redux-mock-store';
import App from './App';

const companies = [
  {
    symbol: 'ATVI',
    name: 'Activision Blizzard',
    sector: 'Communication Services',
    subSector: 'Communication Services',
    headQuarter: 'Santa Monica, CALIFORNIA',
    dateFirstAdded: null,
    cik: '0000718877',
    founded: '1983-06-10',
  },
  {
    symbol: 'ADBE',
    name: 'Adobe',
    sector: 'Technology',
    subSector: 'Technology',
    headQuarter: 'San Jose, CALIFORNIA',
    dateFirstAdded: null,
    cik: '0000796343',
    founded: '1986-01-08',
  },
  {
    symbol: 'ADP',
    name: 'ADP',
    sector: 'Industrials',
    subSector: 'Industrials',
    headQuarter: 'Roseland, NEW JERSEY',
    dateFirstAdded: null,
    cik: '0000008670',
    founded: '1961-09-01',
  },
];

describe('Testing APP component', () => {
  test('Rendering companies', () => {
    const middlewares = [thunk, logger];
    const mockStore = configureStore(middlewares);
    const store = mockStore({ companies });
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
