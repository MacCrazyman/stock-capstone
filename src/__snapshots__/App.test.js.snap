// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing APP component Rendering companies 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App pt-6"
      >
        <header
          class="App-header mx-auto bg-blue flex px-4 fixed top-0 w-full z-10"
        >
          <a
            aria-current="page"
            class="flex w-full active"
            href="/"
          >
            <p>
              &lt;
            </p>
            <h2
              class="m-auto"
            >
               
              Companies
            </h2>
            <span
              class="w-4 h-4 bg-microphone bg-cover my-auto mr-6"
            />
            <span
              class="w-4 h-4 bg-gear bg-cover my-auto"
            />
          </a>
        </header>
        <main
          class="w-full"
        />
        <footer
          class="bg-blue-dark fixed bottom-0 w-full"
        >
          CC License
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App pt-6"
    >
      <header
        class="App-header mx-auto bg-blue flex px-4 fixed top-0 w-full z-10"
      >
        <a
          aria-current="page"
          class="flex w-full active"
          href="/"
        >
          <p>
            &lt;
          </p>
          <h2
            class="m-auto"
          >
             
            Companies
          </h2>
          <span
            class="w-4 h-4 bg-microphone bg-cover my-auto mr-6"
          />
          <span
            class="w-4 h-4 bg-gear bg-cover my-auto"
          />
        </a>
      </header>
      <main
        class="w-full"
      />
      <footer
        class="bg-blue-dark fixed bottom-0 w-full"
      >
        CC License
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
